internal/cli/test.go

```diff
@@ -342,6 +342,33 @@ func (c *cli) appPickerWithCreateOption(ctx context.Context) (pickerOptions, err
 	return enhancedOptions, nil
 }
 
+func (c *cli) audiencePickerOptions(ctx context.Context) (pickerOptions, error) {
+	resourceServerList, err := c.api.ResourceServer.List(ctx, management.PerPage(100))
+	if err != nil {
+		return nil, err
+	}
+
+	var opts pickerOptions
+	for _, resourceServer := range resourceServerList.ResourceServers {
+		label := fmt.Sprintf(
+			"%s %s",
+			resourceServer.GetName(),
+			ansi.Faint(fmt.Sprintf("(%s)", resourceServer.GetIdentifier())),
+		)
+
+		opts = append(opts, pickerOption{
```

Comment:
Should this logic also consider if the API is enabled for the chosen application (for the case of Regular Web Apps and M2M apps), to avoid showing APIs that cannot actually be used with the chosen application?

Reply:
And maybe include a help hint that mentions this when APIs were filtered?

Reply:
Good point. So after a lot of testing the final algorithm is used to determine the audiences presented in the interactive prompt:

- Step 1: Select Client ID
  - Client is M2M
    - Step 2: Select Audience
      - Needs to show api's that are enabled for this client ID
    - Step 3: Scopes
      - Don't shop any input selector.
  - Client is RWA
    - Step2: Select Audience
      - Needs to show api's that are enabled for this client ID
    - Step 3: Scopes
      - Show only the scopes associated with the chosen audience (API)
  - SPA/Native,Generic,SSO Integrations
    - Step 2: Select Audience
      - Accept any audience
    - Step 3: Scopes
      - Show only the scopes associated with the chosen audience (API)